src/lib.rs

```diff
@@ -98,3 +110,15 @@ pub use byteorder::{ByteOrder, BigEndian, LittleEndian};
 #[cfg(feature = "serde")]
 #[doc(hidden)]
 pub mod serde;
+
+/// Custom (internal-only) prelude for this module
```

Comment:
I made this to avoid having to individually import each of these types everywhere they're used.

Seems like a bit of a hack, but dramatically reduces the scope of what the `nightly` feature does which I think is nice.

It also makes this easy to change in the event the [liballoc/libcollections merge is reverted](https://github.com/rust-lang/rust/issues/43112)